hoomd/md/update.py

```diff
@@ -149,133 +150,134 @@ class mueller_plathe_flow:
+    Attention:
+        * This updater uses `hoomd.trigger.Periodic(1)` as a trigger, meaning it
+          is applied every timestep.
         * This updater works currently only with orthorhombic boxes.
 
-    .. note:
-        If you set this updater with unrealistic values, it algorithm might not
-        terminate, because your desired flow target can not be achieved.
 
-    .. versionadded:: v2.1
+    Note:
+        The attributes of this updater are immutable once the updater is
+        attached to a simulation.
 
     Examples::
 
-        #const integrated flow with 0.1 slope for max 1e8 timesteps
-        const_flow = hoomd.variant.linear_interp( [(0,0),(1e8,0.1*1e8)] )
-        #velocity gradient in z direction and shear flow in x direction.
-        update.mueller_plathe_flow(
-            all, const_flow, md.update.mueller_plathe_flow.Z,
-            md.update.mueller_plathe_flow.X, 100)
+        # const integrated flow with 0.1 slope for max 1e8 timesteps
+        ramp = hoomd.variant.Ramp(0.0, 0.1e8, 0, int(1e8))
+        # velocity gradient in z direction and shear flow in x direction.
+        mpf = hoomd.md.update.ReversePerturbationFlow(filter=hoomd.filter.All(),
+                                                      flow_target=ramp,
+                                                      slab_direction="Z",
+                                                      flow_direction="X",
+                                                      n_slabs=20)
+
+        self._flow_target = hoomd.variant._setup_variant_input(flow_target)
+
+    Attributes:
+        filter (hoomd.filter.ParticleFilter): Subset of particles on which to
+            apply this updater.
+
+        flow_target (hoomd.variant.Variant): Integrated target flow in the
+            natural units of the simulation.
+
+        slab_direction (str): Direction perpendicular to the
+            slabs.
+
+        flow_direction (str): Direction of the flow.
+
+        n_slabs (int): Number of slabs.
+
+        max_slab (int): Id < n_slabs where the max velocity component is
+            searched for.
 
+        min_slab (int): Id < n_slabs where the min velocity component is
+            searched for.
     """
 
     def __init__(self,
-                 group,
+                 filter,
                  flow_target,
                  slab_direction,
                  flow_direction,
                  n_slabs,
                  max_slab=-1,
                  min_slab=-1):
-        period = 1  # This updater has to be applied every timestep
+
         assert (n_slabs > 0), "Invalid negative number of slabs."
         if min_slab < 0:
             min_slab = 0
         if max_slab < 0:
             max_slab = n_slabs / 2
+        if max_slab <= -1 or max_slab > n_slabs:
+            raise ValueError("Invalid max_slab in [0," + str(n_slabs) + ").")
+        if min_slab <= -1 or min_slab > n_slabs:
+            raise ValueError("Invalid min_slab in [0," + str(n_slabs) + ").")
```

Comment:
We should include the value of `min_slab` and `max_slab` here, also use f-strings.